mlir/include/mlir/Dialect/GPU/Transforms/Passes.h

```diff
@@ -88,6 +92,7 @@ inline void populateGpuRewritePatterns(RewritePatternSet &patterns) {
   populateGpuAllReducePatterns(patterns);
   populateGpuGlobalIdPatterns(patterns);
   populateGpuShufflePatterns(patterns);
+  populateGpuSubgroupIdPatterns(patterns);
```

Comment:
Let's make sure that this doesn't end up in the SPIR-V lowerings, which seem to have an alternate approach to this op

Reply:
I see. SPV links `get_sub_group_id` function to get the subgroup id. So I think we should just remove this line inside `populateGpuRewritePatterns`.

side note: SPIRV uses the same calculation method to compute subgroup_id.

Reply:
I eventually stripped it from this function.